### PR TITLE
Align font weights and font sizes

### PR DESCRIPTION
**Pull request by janfaracik**
Another small one to align font weights and font sizes, similar to https://github.com/jenkinsci/jenkins/pull/9367 https://github.com/jenkinsci/jenkins/pull/9366 https://github.com/jenkinsci/jenkins/pull/9365.

This MR largely decreases the weight of fonts across Jenkins, ideally increasing legibility and making Jenkins appear more consistent. Font sizes are also adjusted, aligning with whole pixels.

I'll annotate the MR with before/after screenshots.

### Testing done

* Jenkins appears as expected

### Proposed changelog entries

- Align font weights and font sizes.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: war/src/main/scss/abstracts/_theme.scss
Comment (janfaracik) [inline war/src/main/scss/abstracts/_theme.scss:200]: **Before**
image

**After**
image

File: war/src/main/scss/abstracts/_theme.scss
Comment (janfaracik) [inline war/src/main/scss/abstracts/_theme.scss:42]: **Before**
image

**After**
image

File: war/src/main/scss/base/_style.scss
Comment (janfaracik) [inline war/src/main/scss/base/_style.scss:158]: See https://github.com/jenkinsci/jenkins/pull/9393/files#r1640367843

File: war/src/main/scss/components/_dialogs.scss
Comment (janfaracik) [inline war/src/main/scss/components/_dialogs.scss:22]: **Before**
image

**After**
image

File: war/src/main/scss/components/_table.scss
Comment (janfaracik) [inline war/src/main/scss/components/_table.scss:28]: **Before**
image

**After**
image

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
